File: VB/Models/CustomLocalizationHelper.vb
Imports System.Collections.Generic
Imports System.Globalization
Imports System.Threading
Imports System.Web
Imports System.Web.Mvc

Public Class CustomLocalizationHelper
	Public Shared Function GetLanguages() As List(Of SelectListItem)
		Return New List(Of SelectListItem)() From {
			New SelectListItem() With {.Text = "English", .Value = "en", .Selected = True},
			New SelectListItem() With {.Text = "Русский", .Value = "ru"},
			New SelectListItem() With {.Text = "Deutsch", .Value = "de"}
		}
	End Function
	Public Shared Sub ApplyCurrentCulture()
		If HttpContext.Current.Session("CurrentCulture") IsNot Nothing Then
			Dim ci As CultureInfo = DirectCast(HttpContext.Current.Session("CurrentCulture"), CultureInfo)
			Thread.CurrentThread.CurrentUICulture = ci
			Thread.CurrentThread.CurrentCulture = CultureInfo.CreateSpecificCulture(ci.Name)
		End If
	End Sub
End Class

File: CS/t329154/Models/CustomLocalizationHelper.cs
using System.Collections.Generic;
using System.Globalization;
using System.Threading;
using System.Web;
using System.Web.Mvc;

namespace t329154.Models
{
    public class CustomLocalizationHelper
    {
        public static List<SelectListItem> GetLanguages()
        {
            return new List<SelectListItem>() {
                new SelectListItem() { Text = "English", Value = "en", Selected = true },
                new SelectListItem() { Text = "Русский", Value = "ru" },
                new SelectListItem() { Text = "Deutsch", Value = "de" }
            };
        }
        public static void ApplyCurrentCulture()
        {
            if (HttpContext.Current.Session["CurrentCulture"] != null)
            {
                CultureInfo ci = (CultureInfo)HttpContext.Current.Session["CurrentCulture"];
                Thread.CurrentThread.CurrentUICulture = ci;
                Thread.CurrentThread.CurrentCulture = CultureInfo.CreateSpecificCulture(ci.Name);
            }
        }
    }
}